### PR TITLE
Adds dropdown items for New Folder (and requisite work to accomplish this)

### PR DESCRIPTION
**Pull request by dimitropoulos**
There's something rotten in the state of Denmark.  We have a prop drilling problem in our app, which makes it very very difficult to fix things like INS-1568 (that the dropdown menu for folders is lacking the options newly added to the plus button next to the request filter).

While this seems trivial... here's how this one function, `App._createRequest`, is passed around the app:
Screenshot_20220613_144024

Some of these paths are... exhilarating (to say it nicely).  Take the middle one for example:

| Lifetime | Component | Name | Type |
| - | - | - | - |
| 1. | `App` | `_requestCreate` | `(parentId: string, type?: string) => Promise<void>`
| 2. | `Wrapper` | `_handleCreateRequestInWorkspace` | `(type?: string) => void`
| 3. | `WrapperDebug` | `handleRequestCreate` | `() => void`
| 4. | `SidebarFilter` | `requestCreate` | `(type?: string) => void`
| 5. | `SidebarFilter` | `_handleRequestCreate` | `(type?: string) => void`
| 6. | `SidebarCreateDropdown` | `handleCreateRequest` | `(id: string) => any`

It's... a lot to chase after.  And (especially if you examine the types) there are many dead ends (as well as spontaneous generation of new (and wrong!) ends!).

The first thing this PR does is address that by creating a set of higher-level helpers that can use context (via redux) to grab the state they require and continue.  The `selectRootState` function is an unfortunate reality of the NeDB data being "in control" (and not the other way around) otherwise we'd be able to grab whatever we need with thunks.  It's a lot better than the current situation, though.

After this change, here's the exact same state tree from above (green means SFC):
Screenshot_20220613_143325

_____

After that's said and done, it's ABSOLUTELY LAUGHABLY TRIVIAL to add the two items in the dropdown... go figure..

| BEFORE | AFTER |
| - | - |
| Screenshot_20220613_134200 | Screenshot_20220613_134136 |

______

closes INS-1568
changelog(Improvements): The request collection folder context menu now shows options for GraphQL and gRPC (as well as HTTP).

### CODE REVIEW

File: packages/insomnia/src/ui/redux/selectors.ts [hunk 1]
```diff
@@ -21,6 +21,13 @@ type EntitiesLists = {
 // ~~~~~~~~~ //
 // Selectors //
 // ~~~~~~~~~ //
+/**
+ * @deprecated
+ * DO NOT use this unless ABSOLUTELY necessary.
+ * Using this couples state tree shape to components.
+ */
+export const selectRootState = (state: RootState) => state;
```
Comment (jackkav) [inline packages/insomnia/src/ui/redux/selectors.ts]: There surely has to be a better way than adding a selector and immediately deprecating it.

Comment (dimitropoulos) [reply packages/insomnia/src/ui/redux/selectors.ts]: this was removed in [`97df7bb` (#4862)](https://github.com/Kong/insomnia/pull/4862/commits/97df7bb810f4c2b1cbd8f72152533520df86a146)

**Review by marckong — APPROVED** [review]
LGTM!

**Review by filfreire — APPROVED** [review]
LGTM.

There's a slight difference in the text (`New `) we show on a folder and on the `+` (plus) button: 
Screenshot 2022-06-14 at 14 46 01
Screenshot 2022-06-14 at 14 45 53

I would say they should probably look the same, but I leave it up to you 👍

**Comment by dimitropoulos** [comment]
@filfreire the reason is that the context is different.  in one, you're clicking from a dropdown that already has the plus sign and then you have the plus sign next to each item.  in the other, it's a regular context menu.  also, the folder part is a little weird because it says new folder in both (but the icon isn't a plus).  @wdawson feel free to review and let me know if you'd like any copy changes.  easy-peasy, if so.

**Comment by jackkav** [comment]
Thanks @dimitropoulos this one was a wild ride.

**Pull request merged**